src/test/java/teammates/test/cases/webapi/GetStudentProfilePictureActionTest.java

```diff
@@ -29,57 +26,195 @@ protected String getRequestMethod() {
         return GET;
     }
 
-    @BeforeClass
-    public void classSetup() {
-        account = typicalBundle.accounts.get("student1InCourse1");
-        studentProfileAttributes = typicalBundle.profiles.get("student1InCourse1");
-    }
-
     @Override
     @Test
     public void testExecute() throws Exception {
-        testActionWithNoParams();
-        testActionWithBlobKey();
-    }
 
-    private void testActionWithNoParams() {
+        ______TS("Success case: student gets his own image");
 
-        ______TS("Failure case: no parameters given");
-        loginAsStudent(account.googleId);
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        loginAsStudent(student1InCourse1.googleId);
 
-        String[] submissionParams = new String[] {};
+        GetStudentProfilePictureAction action = getAction();
+        ImageResult imageResult = getImageResult(action);
 
-        GetStudentProfilePictureAction action = getAction(submissionParams);
-        assertThrows(NullHttpParameterException.class, () -> action.execute());
-    }
+        assertEquals(HttpStatus.SC_OK, imageResult.getStatusCode());
```

Comment:
You should verify that the image result is correct instead of just the status code?
(Same scenarios for all of your other assertEquals)

Reply:
Yup done!